packages/core/src/hydration/error_handling.ts

```diff
@@ -60,21 +62,23 @@ export function validateMatchingNode(
     const hostComponentDef = getDeclarationComponentDef(lView);
     const componentClassName = hostComponentDef?.type?.name;
 
-    const expected = `Angular expected this DOM:\n\n${
-        describeExpectedDom(lView, tNode, isViewContainerAnchor)}\n\n`;
+    const expectedDom = describeExpectedDom(lView, tNode, isViewContainerAnchor);
+    const expected = `Angular expected this DOM:\n\n${expectedDom}\n\n`;
 
     let actual = '';
-
     if (!node) {
       // No node found during hydration.
       header += `the node was not found.\n\n`;
+      markRNodeAsMismatchedByHydration(lView[HOST]!, expectedDom);
```

Comment:
We'll need to use `unwrapRNode` here to account for `LContainer`s as hosts:
```suggestion
      markRNodeAsMismatchedByHydration(unwrapRNode(lView[HOST])!, expectedDom);
```

Reply:
Could we please also add a quick comment here to explain why we use `lView[HOST]` here?

(there is also another usage of `lView[HOST]` below, so both comments are also relevant there as well)